code-challenges.test.js

```diff
@@ -14,45 +14,128 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
 
-// b) Create the function that makes the test pass.
+// describe("codedMessage", () => {
+//     it("returns a string with a coded message.", () => {
+//         const secretCodeWord1 = "Lackadaisical"
+//             // Expected output: "L4ck4d41s1c4l"
+//         const secretCodeWord2 = "Gobbledygook"
+//             // Expected output: "G0bbl3dyg00k"
+//         const secretCodeWord3 = "Eccentric"
+//             // Expected output: "3cc3ntr1c"
+//         expect(codedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+//         expect(codedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+//         expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+//     })
+// })
 
-// Pseudo code:
 
-// --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
-// a) Create a test with expects statement using the variable provided.
+// // b) Create the function that makes the test pass.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+// // Pseudo code:
+ 
+// // 1.) Declare a function with a parameter(string)
 
-const filterLetterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const filterLetterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
+// const codedMessage = (string) => {
+
+// // 2.) After creating the function, we need to find a built-in method that can change the requested values in any string we add to this function.
+
+// return string.replaceAll("a", "4").replaceAll("e", "3").replaceAll("i", "1").replaceAll("o", "0").replaceAll("E", "3");
+
+// // I would put more time into not hard coding the capitol "E" and to have a dynamic code to where if I put a character in it'll change whether capitolized or not.
```

Comment:
You can look into regex to make that happen. For the pseudo code, I would explain what the code is actually doing instead of saying "we need to find a built-in method..."